xcom2-launcher/xcom2-launcher/Classes/Mod/ModList.cs

```diff
@@ -205,66 +206,66 @@ internal void UpdateMod(ModEntry m, Settings settings)
                     m.SetSource(ModSource.SteamWorkshop);
 
                 else
-                // in workshop path but not loaded via steam
+                    // in workshop path but not loaded via steam
                     m.SetSource(ModSource.Manual);
             }
-            
+
             // Ensure source ID exists
             if (m.WorkshopID <= 0)
             {
-                long sourceID;
-
-                if (m.Source == ModSource.SteamWorkshop && long.TryParse(Path.GetFileName(m.Path), out sourceID))
+                if (m.Source == ModSource.SteamWorkshop && long.TryParse(Path.GetFileName(m.Path), out var sourceID))
                 {
                     m.WorkshopID = sourceID;
                 }
                 else
                 {
                     m.WorkshopID = new ModInfo(m.GetModInfoFile()).PublishedFileID;
                 }
-
             }
-            
+
             // Fill Date Added
             if (!m.DateAdded.HasValue)
                 m.DateAdded = DateTime.Now;
 
+            SteamUGCDetails_t workshopDetails = new SteamUGCDetails_t();
 
             // Check Workshop for infos
             if (m.WorkshopID != 0)
             {
-                var publishedID = (ulong) m.WorkshopID;
-
-                var value = Workshop.GetDetails(publishedID, string.IsNullOrEmpty(m.Description));
+                workshopDetails = Workshop.GetDetails((ulong) m.WorkshopID, string.IsNullOrEmpty(m.Description));
+            }
 
+            if (workshopDetails.m_eResult == EResult.k_EResultOK)
```

Comment:
You sure this condition will evaluate to false for failed requests and local mods?

Reply:
Yes, m_eResult is initialized to 0 (which is actually a non-existent EResult value) when SteamUGCDetails_t is created.

Reply:
> which is actually a non-existent EResult value

Right, this is the part that got me confused. I thought it would initialize to the first enum value (which is `k_EResultOK`)